### PR TITLE
Improvement: Add uppercase support for fromExtension

### PR DESCRIPTION
**Pull request by keremnalbant**
* Add uppercase support for fromExtension
* Add regarding tests

https://answers.microsoft.com/en-us/windows/forum/all/file-extensions-capitalized/c31e48ef-ab94-40ed-b55f-0ebe744e2f4d

On some Windows machines, file extensions are auto-capitalized. I thought it would be great to handle this on the library if it's passed as uppercase

### CODE REVIEW

**Comment by keremnalbant** [comment]
https://answers.microsoft.com/en-us/windows/forum/all/file-extensions-capitalized/c31e48ef-ab94-40ed-b55f-0ebe744e2f4d

On some Windows machines, file extensions are auto-capitalized. I thought it would be great to handle this on our end

File: src/ext.test.ts [hunk 1]
```diff
@@ -2,143 +2,110 @@ import { FileKind } from './data';
...
-        ['webp', 'image'],
-        ['gif', 'image'],
-      ])('%s should return %s', function (input, expected) {
+      const extensions = ['png', 'jpg', 'jpeg', 'svg', 'webp', 'gif'];
+      const testCases = getTestCases(extensions, 'image');
+      test.each(testCases)('%s should return %s', function (input, expected) {
         expect(fromExtension(input)).toBe(expected);
         expect(fromExtension(`.${input}`)).toBe(expected);
       });
     });
```
Comment (anttiviljami) [inline src/ext.test.ts:22]: Not a fan of this change. This makes tests harder to read and understand.

**Review by anttiviljami — CHANGES_REQUESTED** [review]

**Comment by anttiviljami** [comment]
I don't like the change to make tests more clever.

Otherwise looks ok to me.

File: src/ext.test.ts [hunk 2]
```diff
@@ -2,143 +2,110 @@ import { FileKind } from './data';
 import { fromExtension, getExtensions } from './ext';
 
 describe('extension', () => {
+  const getTestCases = (extensions: string[], type: string) => {
+    const testCases: Array<[string, string]> = [];
+    extensions.forEach((ext) => {
+      testCases.push([ext, type]);
+      testCases.push([ext?.toUpperCase(), type]);
```
Comment (anttiviljami) [inline src/ext.test.ts:9]: Please don't hide this here. There should be an explicit test to describe the behaviour that uppercase extensions are supported.

**Pull request merged**